base/sparse/linalg.jl

```diff
@@ -532,11 +531,14 @@ function normestinv{T}(A::SparseMatrixCSC{T}, t::Integer = 2)
     end
     ind = Array(Int64, n)
     ind_hist = Array(Int64, maxiter * t)
-    S = zeros(T, n, t)
+
+    Ti = typeof(float(zero(T)))
```

Comment:
`S` is not always integer valued.  It is integer valued when `T` is Real, and it is complex otherwise -- that was  the problem I was running into after I added tests.

I can add a different type variable for `S` to catch that corner case.